src/test/kotlin/org/rust/ide/inspections/RsNeedlessLifetimesInspectionTest.kt

```diff
@@ -281,6 +281,20 @@ class RsNeedlessLifetimesInspectionTest : RsInspectionsTestBase(RsNeedlessLifeti
         }
     """)
 
+    fun `test no elision for impl 1`() = doTest("""
```

Comment:
Looks like your tests don't really work, because std isn't linked. Consider add an annotation `@ProjectDescriptor(WithStdlibRustProjectDescriptor::class)` to these tests or write them without using of standard traits